libunwind/src/Registers.hpp

```diff
@@ -4554,10 +4552,21 @@ class _LIBUNWIND_HIDDEN Registers_riscv {
   void        setSP(reg_t value) { _registers[2] = value; }
   reg_t       getIP() const { return _registers[0]; }
   void        setIP(reg_t value) { _registers[0] = value; }
+  reg_t       getFP() const { return _registers[8]; }
+  void        setFP(reg_t value) { _registers[8] = value; }
```

Comment:
Eh, I would argue there's no need to encode FP, it's just a saved register, and may not be being used as a frame pointer. Then you don't need to extend each Registers_arch with knowledge about the FP.

Reply:
The trusted stack frame treats `fp` as a special register and does not store it in the array of callee-saved registers. So without the setter, I'd need some other code that does the equivalent thing, which complicates `unwindIfAtBoundary`.

I suppose we can gate the getter/setter by `_LIBUNWIND_HAS_CHERI_LIB_C18N`?

Reply:
I mean yes, part of my point is that the interface should really have put fp in regs. But what's wrong with `setRegister(UNW_ARCH_[XC]N, state.fp)` for now until we change the interface to be more sensible? You already need to list the `UNW_ARCH_[XC]N` for all the other callee-saved registers.

Reply:
That works for me. The reason fp is separately stored in the trusted frame is so that the trusted frame would 'look like' a normal frame to the kernel unwinder. When a special flag is passed to RTLD, the trampoline will not clear the tag of fp when entering a callee compartment, allowing anyone to follow fp to cross compartment boundaries and trace whole call stack.

This should be useful when using pmcstat, but we have never actually tried it...